### PR TITLE
E_CLASSROOM-204 [FE/BE] Make Category Quizzes page dynamic

### PR DESCRIPTION
**Pull request by CaneteJoash**
### Issue Link
https://framgiaph.backlog.com/view/E_CLASSROOM-204
### Definition of Done
- [x] User will be able to see attempts of quiz dynamically

### Commands to Run
N/A

#### Main note

I have backtend pr.
- link: https://github.com/framgia/sph-classroom-els-be/pull/39
#### Pages Affected
- http://localhost:3003/categories/1/quizzes

### Scenarios/ Test Cases
- [x] User will be able to see attempts of quiz dynamically
#### User Interface
N/A
#### User Experience 
N/A
#### Functionality
N/A
### Screenshots
image

### CODE REVIEW

**Comment by Johnpaullim** [comment]
### Working 

- [x] User able to see attempts of quiz dynamically
- [x] Able to Improvement the category quizzes page dynamic

<details><summary>Screenshot</summary>

Screenshot 2021-12-14 182317

Screenshot 2021-12-14 182640

Screenshot 2021-12-14 182748

</details> 

- Total Test time: 0.5 hr

File: src/pages/student/quizzes/QuizList/components/QuestionGrid/index.js [hunk 1]
```diff
@@ -1,34 +1,66 @@
...
+  const getTotalTimeLimit = () => {
+    if (questions != null) {
+      let total = 0;
+
+      for (let x = 0; x < questions.length; x++) {
+        total += questions[x].time_limit;
+      }
+
+      return total;
+    }
```
Comment (RunstyleX) [inline src/pages/student/quizzes/QuizList/components/QuestionGrid/index.js]: try to look into es6 "reduce" function for arrays / array of objects and see if you can apply it here.

File: src/api/QuizTakenReview/index.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import API from '../Base';
+
+const QuizzesTakenReviewApi = {
+  getAll: (quiz_id) => {
+    const options = {
+      method: 'GET',
+      url: `/QuizeAttempts/${quiz_id}`,
```
Comment (RunstyleX) [inline src/api/QuizTakenReview/index.js]: Please fix typo

File: src/pages/student/quizzes/QuizList/components/QuestionGrid/index.js [hunk 3]
```diff
@@ -46,41 +77,33 @@ const QuestionGrid = ({ quiz }) => {
...
+            </p>
           </div>
-        )}
+          <div className={style.ResultScore} style={{ fontWeight: 'bold' }}>
+            <p>Latest Score</p>
+            <p>
+              {getLatestScore() >= 0 ? getLatestScore() : 0}/{questions?.length}
+            </p>
+          </div>
+          {/* {`${quiz.correctAnswers}/${quiz.totalQuestions}`} */}
```
Comment (RunstyleX) [inline src/pages/student/quizzes/QuizList/components/QuestionGrid/index.js]: Remove if not needed anymore

File: src/api/QuizTakenReview/index.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import API from '../Base';
+
+const QuizzesTakenReviewApi = {
+  getAll: (quiz_id) => {
+    const options = {
+      method: 'GET',
+      url: `/QuizAttempts/${quiz_id}`,
```
Comment (RunstyleX) [inline src/api/QuizTakenReview/index.js]:
```suggestion
      url: `/quiz_attempts/${quiz_id}`,
```

**Pull request merged**